spec/datadog/di/code_tracker_spec.rb

```diff
@@ -0,0 +1,119 @@
+require "datadog/di/code_tracker"
+
+RSpec.describe Datadog::DI::CodeTracker do
+  let(:tracker) do
+    described_class.new
+  end
+
+  describe ".new" do
+    it "creates an instance" do
+      expect(tracker).to be_a(described_class)
+    end
+  end
+
+  describe "#start" do
+    after do
+      tracker.stop
+    end
+
+    it "tracks loaded files" do
+      # The expectations appear to be lazy-loaded, therefore
+      # we need to invoke the same expectation before starting
+      # code tracking as we'll be using later in the test.
+      expect(tracker.send(:registry)).to be_empty
+      tracker.start
+      # Should still be empty here.
+      expect(tracker.send(:registry)).to be_empty
+      load File.join(File.dirname(__FILE__), "code_tracker_test_class_1.rb")
+      expect(tracker.send(:registry).each.to_a.length).to eq(1)
```

Comment:
Can we do just length? Since we are looking into guts, we assume that it behave like "something enumerable", do we need to wrap it as one and make it also array?

```suggestion
      expect(tracker.send(:registry).length).to eq(1)
```

Reply:
Concurrent::Map apparently doesn't have a length and requires this incantation, but now that I switched to a plain Hash I will get rid of it. Thanks for spotting.